core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -502,7 +503,7 @@ public boolean delete(Object key) {
       if (loadMetadata && fe.metadataLen > 0) {
          metadataBb = factory.newByteBuffer(data, fe.keyLen + fe.dataLen, fe.metadataLen);
       }
-      return entryFactory.create(keyBb, valueBb, metadataBb);
+      return entryFactory.create(keyBb, valueBb, metadataBb, fe.expiryTime, fe.lastUsed);
```

Comment:
Related: For SIFS I have changed it so that the created and lastUsed timestamps are not stored in the EntryHeader, instead they are stored as part of the record if metadata is present/required.